### DISCUSSION TITLE
FeatureNode Line Updating

### DISCUSSION THREAD

**Discussion by Warren-Whitfield** [pelicanmapping/osgearth#2574] [Support Q&A] [answered]
If I have a simple line with two points only, like this...
```
pLine = new LineString();
pLine->push_back(...);
pLine->push_back(...);
pLineFeature = new Feature(pLine, m_pGeoSRS);
pLineFeatureNode = new FeatureNode(pLineFeature, m_velVectorLineStyle);
m_pVelocityVectorGroup->addChild(pLineFeatureNode);
```
Is this the best way to update the line without having to re-create it?
```
pLine->clear();
pLine->push_back(...);
pLine->push_back(...);
pLineFeatureNode->dirty();
```
Basically I have a line that needs to update periodically, but I'm getting a crash at random times and it looks like its in this area.

**Comment by gwaldron** [comment, marked as answer]
Make sure you are updating it safely, i.e. during the update traversal and not from a background thread or anything.
Do you have a stack trace?

**Comment by WarrenWhitfield** [comment]
I suspect that may have been the problem (updating outside of the traversal)
I modified the update to now be in an installed callback, so I'll monitor and see what happens. So far so good though.
If it happens again, I'll capture the stack trace, but from what I recall, it did in fact have something to do with nodeVisitor, traversal or something of the like.
